util/annotations/annotations.go

```diff
@@ -38,14 +38,24 @@ func New() *Annotations {
 
 // Add adds an annotation (modeled as a Go error) in-place and returns the
 // modified Annotations for convenience.
-func (a *Annotations) Add(err error) Annotations {
+func (a *Annotations) Add(err annoErr) Annotations {
```

Comment:
Again, it feels weird to expose a non-exported type in an exported method. The additional `AddRaw` also creates a lot of friction for the caller. Can we leave the old API intact (just with the one `Add(error)` method) and do the right thing behind the scenes with type assertions and `error.As`? (Does the latter work with interfaces?)